src/Hl7.Fhir.Base/FhirPath/Expressions/Closure.cs

```diff
@@ -82,5 +98,13 @@ public virtual IEnumerable<ITypedElement> ResolveValue(string name)
 
             return null;
         }
+
+        private static ITypedElement getResourceFromNode(ScopedNode node) => node.AtResource ? node : node.ParentResource;
+        
+        private static ITypedElement getRootResourceFromNode(ScopedNode node)
+        {
+            var resource = (ScopedNode)getResourceFromNode(node);
+            return resource?.ParentResource?.ContainedResources() is {} c && c.Any() ? resource.ParentResource : resource;
```

Comment:
Doesn't this draw the wrong conclusion if we have a DomainResource with child resources (not contained, but just an element of typeResource), and if we are such a child resource?  Then our boundary shifts up, if though it should not?  I think we should check that the parent node of us (as a resource) is called "contained" instead.